### PR TITLE
Update to `check-release@v2`

### PR DESCRIPTION
**Pull request by jtpio**
Investigate releaser workflow failures: https://github.com/jupyterlite/xeus-python-kernel/actions/runs/4185862759

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -15,3 +15,13 @@ npm = ["jlpm"]
 
 [tool.check-manifest]
 ignore = ["share/jupyter/labextensions/@jupyterlite/xeus-python-kernel/**", "yarn.lock", ".*", "package-lock.json", "src/xpython_wasm.js", "src/xpython_wasm.wasm", "src/*.data", "src/*.js", "src/worker.ts", "src/web_worker_kernel.ts", "*.sh"]
+
+[tool.jupyter-releaser.options]
+skip = ["check-links"]
+
+[tool.jupyter-releaser.hooks]
+before-build-npm = [
+    "python -m pip install jupyterlab~=3.1 empack>=2.0.9,<3 jupyter_packaging~=0.10,<2",
```
Comment (jtpio) [inline pyproject.toml:24]: This is probably because the `v1` used to install the Python package in editable mode before with `pip install -e .`

**Comment by jtpio** [comment]
I think releaser v2 has an issue with not reading the config from `package.json` if `pyproject.toml` exists.

But this PR moves the config to `pyproject.toml` anyway and the check now passes.

**Comment by jtpio** [comment]
Maybe a follow-up could be to migrate to `hatch` if we want to streamline the setup more.

**Comment by jtpio** [comment]
I'll proceed with the merge and follow-up with a new release to double check it fixes it.

**Comment by martinRenou** [comment]
Thanks!

**Pull request merged**